python-meh.spec

```diff
@@ -20,47 +17,23 @@ License: GPLv2+
 Group: System Environment/Libraries
 BuildArch: noarch
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
-BuildRequires: python-devel
 BuildRequires: gettext
-BuildRequires: python-setuptools
 BuildRequires: intltool
-BuildRequires: dbus-python
 BuildRequires: libreport-gtk >= %{libreportver}
 BuildRequires: libreport-cli >= %{libreportver}
-BuildRequires: libreport-python >= %{libreportver}
-BuildRequires: python-six
 
-%if 0%{with_python3}
 BuildRequires: python3-devel
 BuildRequires: python3-setuptools
 BuildRequires: python3-dbus
 BuildRequires: libreport-python3 >= %{libreportver}
 BuildRequires: python3-six
-BuildRequires: python3-pocketlint
-%endif
-
-Requires: python
-Requires: dbus-python
-Requires: rpm-python
-Requires: libreport-cli >= %{libreportver}
-Requires: libreport-python >= %{libreportver}
-Requires: python-six
 
-%description
-The python-meh package is a python library for handling, saving, and reporting
+%global _description\
+The python-meh package is a python library for handling, saving, and reporting\
 exceptions.
 
-%package gui
-Summary: Graphical user interface for the python-meh library
-Requires: python-meh = %{version}-%{release}
-Requires: pygobject3
-Requires: gtk3
-Requires: libreport-gtk >= %{libreportver}
-
-%description gui
-The python-meh-gui package provides a GUI for the python-meh library.
+%description %_description
 
-%if 0%{with_python3}
```

Comment:
You should add the `%python_provide` [macro](https://fedoraproject.org/wiki/Packaging:Python#Provides).

So it should be like:
```
%package -n python3-%{srcname}
...
%{?python_provide:%python_provide python3-%{srcname}}
...
```

Similar for the `python3-meh-gui` subpackage.

Reply:
That looks a bit too cryptic to me, let's keep the current way for now.

Reply:
Add it. Also, add obsoletes for python- and python2- versions otherwise you are breaking the upgrade path.

Reply:
What version should I put into the Obsoletes ? The last version which had the Python  packages ? That would look like this:

    Obsoletes: python-meh <= 0.45
    Obsoletes: python2-meh <= 0.45

Reply:
Somebody told me it should be `< version-release` with first version-release that no longer has them (i.e. don't use `<=` because of black magic).

Also, correction from my previous comment, `%{?python_provide:%python_provide python3-%{srcname}}` does not add the obsoletes just yet. Add it anyway, it will add provides in the future.

Reply:
Interesting - so it should be: 

    Obsoletes: python-meh < 0.46-1
    Obsoletes: python2-meh < 0.46-1